src/main.rs

```diff
@@ -197,14 +197,16 @@ fn run() -> Result<(), Error> {
     )?);
     report.push_result(execute(|| node::yarn_global_update(run_type), opt.no_retry)?);
 
-    #[cfg(not(any(
-        target_os = "freebsd",
-        target_os = "openbsd",
-        target_os = "netbsd",
-        target_os = "dragonfly"
-    )))]
-    report.push_result(execute(|| generic::run_apm(run_type), opt.no_retry)?);
-    report.push_result(execute(|| generic::run_gem(&base_dirs, run_type), opt.no_retry)?);
+    if !opt.no_ruby {
+        #[cfg(not(any(
+            target_os = "freebsd",
+            target_os = "openbsd",
+            target_os = "netbsd",
+            target_os = "dragonfly"
+        )))]
+        report.push_result(execute(|| generic::run_apm(run_type), opt.no_retry)?);
```

Comment:
APM is the Atom Package Manager. It's unrelated to Ruby.

Reply:
I suspected this was the case, but figured out I'd wait for review before I push any more changes.

Will implement these later today.